rasterio/_warp.pyx

```diff
@@ -659,24 +719,71 @@ cdef class WarpedVRTReaderBase(DatasetReaderBase):
         self._gcps = None
         self._read = False
 
+        # Deprecation of "dst" parameters.
+        if dst_nodata is not None:
+            warnings.warn(
+                "dst_nodata will be removed after 1.0, use nodata",
+                RasterioDeprecationWarning)
+
+        if nodata is None:
+            nodata = dst_nodata
```

Comment:
wonder if this line should be inside the ` if dst_nodata is not None:` block ?

Reply:
It could, but I want to keep it outside for now. We'll be deleting all this deprecation-related code soon. I'll make a note of that in the function.